autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -550,105 +322,52 @@ def stack_new_level_core(self, X, y, X_val=None, y_val=None, models=None, level=
-                                     num_classes=self.num_classes, save_bagged_folds=save_bagged_folds, random_state=level+self.random_seed)
+                stacker_type(
+                    path=self.path, name=model.name + '_STACKER_l' + str(level), model_base=model, base_model_names=base_model_names,
+                    base_model_paths_dict=base_model_paths, base_model_types_dict=base_model_types, use_orig_features=use_orig_features,
+                    num_classes=self.num_classes, save_bagged_folds=save_bagged_folds, random_state=level+self.random_seed
+                )
                 for model in models]
-        X_train_init = self.get_inputs_to_stacker(X, level_start=0, level_end=level, fit=True)
+        X_train_init = self.get_inputs_to_stacker(X, base_models=base_model_names, fit=True)
         if X_val is not None:
-            X_val = self.get_inputs_to_stacker(X_val, level_start=0, level_end=level, fit=False)
-
-        return self.train_multi(X_train=X_train_init, y_train=y, X_val=X_val, y_val=y_val, models=models, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, level=level, stack_name=stack_name, kfolds=kfolds, n_repeats=n_repeats, time_limit=time_limit)
+            X_val = self.get_inputs_to_stacker(X_val, base_models=base_model_names, fit=False)
 
-    def stack_new_level_aux(self, X, y, level, fit=True, time_limit=None):
-        stack_name = 'aux1'
-        X_train_stack_preds = self.get_inputs_to_stacker(X, level_start=0, level_end=level, fit=fit)
-        return self.generate_weighted_ensemble(X=X_train_stack_preds, y=y, level=level, kfolds=0, n_repeats=1, stack_name=stack_name, time_limit=time_limit)
+        return self._train_multi(X_train=X_train_init, y_train=y, X_val=X_val, y_val=y_val, models=models, hyperparameter_tune=hyperparameter_tune, feature_prune=feature_prune, level=level, stack_name=stack_name, k_fold=k_fold, n_repeats=n_repeats, time_limit=time_limit)
 
-    def generate_weighted_ensemble(self, X, y, level, kfolds=0, n_repeats=1, stack_name=None, hyperparameters=None, time_limit=None, base_model_names=None, name_suffix='', save_bagged_folds=None, check_if_best=True, child_hyperparameters=None):
-        if save_bagged_folds is None:
-            save_bagged_folds = self.save_bagged_folds
-        if base_model_names is None:
-            base_model_names = self.models_level['core'][level - 1]
-        if len(base_model_names) == 0:
-            logger.log(20, 'No base models to train on, skipping weighted ensemble...')
-            return []
+    # TODO: Consider making level be auto-determined based off of max(base_model_levels)+1
+    # TODO: Remove name_suffix, hacked in
+    def stack_new_level_aux(self, X, y, base_model_names: List[str], level, fit=True, stack_name='aux1', time_limit=None, name_suffix='') -> List[str]:
+        """
+        Trains auxiliary models (currently a single weighted ensemble) using the provided base models.
```

Comment:
clarify what makes model (like weighted ensemble) auxiliary instead of core

Reply:
added